kdump/include/runtest.sh

```diff
@@ -0,0 +1,595 @@
+            # memory is less the threshold required by kdump service.
+            /bin/systemctl enable kdump.service || /sbin/chkconfig kdump on
+
+            Log "- Changing boot loader."
+            {
+                /sbin/grubby                     \
+                    --args="${KER1ARGS}"         \
+                    --update-kernel="${default}" &&
+                if [ "${K_ARCH}" = "s390x" ]; then zipl; fi
+            } || FatalError "Error changing boot loader."
+
+            Report 'pre-reboot'
+            Log "- Rebooting\n"; sync; rhts-reboot
+        }
+    fi
+
+    Log "- kexec-tools kernel versions"
+    rpm -q kexec-tools kernel
+
+    Log "- Crashkernel reservation and current cmdline"
+    rpm -q lshw || yum -y install lshw
+    echo "Total system memory: $(lshw -short | grep -i "System Memory" | awk '{print $3}')"
+    cat /proc/cmdline
+    kdumpctl showmem || cat /sys/kernel/kexec_crash_size
+    grep "fadump=on" /proc/cmdline && dmesg | grep "firmware-assisted dump" | grep "Reserved"
+    [ -f "${K_REBOOT}" ] && rm -f "${K_REBOOT}"
+
+    # Make sure kdump service fully up after a boot
+    # If kdump service is not started yet, wait for max 5 mins.
+    # It may take time to start kdump service.
+    Log "- Waiting for kdump service to be fully up"
+    local kdump_status=off
+    for i in {1..5}
+    do
+        kdumpctl status 2>&1 || service kdump status 2>&1
+        [ $? -eq 0 ] && {
+            kdump_status=on
+            break
+        }
+        sleep 60
+    done
+
+    # Reset kdump config andd restart kdump service
+    Log "- Reset to default kdump config and restart kdump service"
+    ResetKdumpConfig
+    ReportKdumprd
+    # last try to make sure kdump is ready.
+    sleep 10
+}
+
+DefKdumpMem()
+{
+    local args=""
+
+    if $IS_RHEL6; then
+        if   [[ "${K_ARCH}" == i?86     ]]; then args="crashkernel=128M"
+        elif [[ "${K_ARCH}"  = "x86_64" ]]; then args="crashkernel=128M"
+        elif [[ "${K_ARCH}"  = "ppc64"  ]]; then args="crashkernel=256M"
+        elif [[ "${K_ARCH}"  = "s390x"  ]]; then args="crashkernel=128M"
+        fi
+
+    elif $IS_RHEL7; then
+        if   [[ "${K_ARCH}"  = "x86_64" ]]; then args="crashkernel=160M"
+        elif [[ "${K_ARCH}"  = "s390x"  ]]; then args="crashkernel=160M"
+        elif [[ "${K_ARCH}"  = ppc64*  ]]; then
+            args="crashkernel=0M-4G:384M,4G-16G:512M,16G-64G:1G,64G-128G:2G,128G-:4G"
+        elif [[ "${K_ARCH}"  = "aarch64"  ]]; then args="crashkernel=512M"
+        fi
+
+    elif $IS_RHEL8; then
+        if   [[ "${K_ARCH}"  = "x86_64" ]]; then args="crashkernel=160M"
+        elif [[ "${K_ARCH}"  = "s390x"  ]]; then args="crashkernel=160M"
+        elif [[ "${K_ARCH}"  = ppc64*  ]]; then
+            args="crashkernel=0M-4G:384M,4G-16G:512M,16G-64G:1G,64G-128G:2G,128G-:4G"
+        elif [[ "${K_ARCH}"  = "aarch64"  ]]; then args="crashkernel=512M"
+        fi
+
+    elif $IS_RHEL5; then
```

Comment:
Checking all the `if`s here, are we planning to run this task on Fedora kernels?

Reply:
@veruu I think it's ok if we do run on the Fedora kernel since the base compose is rhel8, could we go ahead and merge and make changes later if needed ?

Reply:
Sure, that's fair